### PR TITLE
[PBL-519] Subsquid: integrate new coingecko endpoint to query multiple historical prices

### PR DESCRIPTION
**Pull request by andresmechali**
https://app.clickup.com/t/20465559/PBL-519

### CODE REVIEW

**Comment by KaiserKarel** [comment]
Task linked: [PBL-519 Subsquid: Implement new Coingecko endpoint for querying all price history at once](https://app.clickup.com/t/20465559/PBL-519)

File: subsquid/src/dbHelper.ts [hunk 1]
```diff
@@ -529,78 +524,87 @@ export async function getCurrentAssetPrices(ctx: Context | EntityManager): Promi
...
   }
 
-  // If any price is missing, get all from Coingecko and update DB
-  if (missingPrices) {
-    const queryIds = assetList
-      .map(({ coingeckoId }) => coingeckoId)
+  // If there are assets for updating, query Coingecko
+  if (updatePrices.size > 0) {
+    // Create query url
+    const queryIds = Array.from(updatePrices)
```
Comment (bengalmozzi) [inline subsquid/src/dbHelper.ts:560]: Is Array.from needed here? Imo updatePrices.map is enough

**Review by easteregg — APPROVED** [review]

File: subsquid/src/dbHelper.ts [hunk 2]
```diff
@@ -841,6 +845,87 @@ export async function getHistoricalCoingeckoPrice(
...
+    throw new Error(`Asset ${assetId} not found in asset list.`);
+  }
+  let coinId: string;
+  if (assetInfo.coingeckoId) {
+    coinId = assetInfo.coingeckoId;
+  } else if (assetInfo.spotPriceBaseAsset) {
+    coinId = assetInfo.spotPriceBaseAsset.coingeckoId;
+  } else {
+    throw new Error("No Coingecko ID found");
+  }
```
Comment (bengalmozzi) [inline subsquid/src/dbHelper.ts:863]: L856-863 can be minimized: 

const coinId = assetInfo.coingeckoId || assetInfo.spotPriceBaseAset?.coingeckoId
if (!coinId) {
    throw new Error("No Coingecko ID found");
}

File: subsquid/src/processors/pablo.ts [hunk 3]
```diff
@@ -319,37 +323,38 @@ export async function processLiquidityRemovedEvent(ctx: Context, block: Block, e
...
+
+      asset.totalLiquidity -= amount;
+      asset.blockId = block.header.hash;
+
+      await ctx.store.save(asset);
+
+      const historicalLockedValue = new HistoricalLockedValue({
+        id: randomUUID(),
+        event,
+        amount: -amount,
```
Comment (bengalmozzi) [inline subsquid/src/processors/pablo.ts:340]: How about considering it to write `amount * -1` ? (It has the same result, just improve the readability.

**Review by bengalmozzi — APPROVED** [review]
Just trivial comments, otherwise LGTM

File: subsquid/src/dbHelper.ts (hunk 1 above)
Comment (andresmechali) [reply subsquid/src/dbHelper.ts:560]: It's a `Set` (to avoid duplications) so it does not have the `.map` method

Comment (bengalmozzi) [reply subsquid/src/dbHelper.ts:560]: Oh right.

Comment (andresmechali) [reply subsquid/src/dbHelper.ts:560]: A valid alternative is `[...updatePrices].map` but it's essentially the same and maybe more confusing

File: subsquid/src/dbHelper.ts (hunk 2 above)
Comment (andresmechali) [reply subsquid/src/dbHelper.ts:863]: True. Will update.

File: subsquid/src/processors/pablo.ts (hunk 3 above)
Comment (andresmechali) [reply subsquid/src/processors/pablo.ts:340]: Done 👌🏼

**Pull request merged**